cql3/expr/expression.hh

```diff
@@ -155,8 +175,14 @@ public:
     expression& operator=(const expression&);
     expression& operator=(expression&&) noexcept = default;
 
-    friend auto visit(invocable_on_expression auto&& visitor, const expression& e);
-    friend auto visit(invocable_on_expression_ref auto&& visitor, expression& e);
+    template<invocable_on_const_expression_ref Visitor>
+    friend std::result_of_t<Visitor(const conjunction&)> visit(Visitor&& visitor, const expression& e);
```

Comment:
I think `decltype(auto)` might also work. This would prevent the need to split invocable_on_expression.

Though the split may be good for other reasons later.

Reply:
You're right, `decltype(auto)` works too and it's a much smaller change, I'm gonna change the code to use it.